### PR TITLE
refactor(eslint-mc-app): merge CRA config to avoid possible ESLint merge errors

### PR DESCRIPTION
**Pull request by emmenko**
Recently we've been seeing this kind of errors when we had to update dependencies.

image

To try avoiding this as much as possible, I thought it would make sense to remove the `eslint-config-react-app` dependency and include most of its configuration into our own config.

So most of the things are copied over.

### CODE REVIEW

File: packages/eslint-config-mc-app/helpers/rules-presets.js [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+ * This file contains rules copied from other libraries, to make it easier to differentiate with
+ * our own rules and to better maintain them.
+ */
+
+// The ESLint browser environment defines all browser globals as valid,
+// even though most people don't know some of them exist (e.g. `name` or `status`).
+// This is dangerous as it hides accidentally undefined variables.
+// We blacklist the globals that we deem potentially confusing.
+// To use them, explicitly reference them, e.g. `window.name` or `window.status`.
+const restrictedGlobals = require('confusing-browser-globals');
```
Comment (CarlosCortizasCT) [inline packages/eslint-config-mc-app/helpers/rules-presets.js:11]: 💯

**Review by CarlosCortizasCT — APPROVED** [review]
👍

File: packages/eslint-config-mc-app/helpers/rules-presets.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
+/**
+ * This file contains rules copied from other libraries, to make it easier to differentiate with
+ * our own rules and to better maintain them.
+ */
+
+// The ESLint browser environment defines all browser globals as valid,
+// even though most people don't know some of them exist (e.g. `name` or `status`).
+// This is dangerous as it hides accidentally undefined variables.
+// We blacklist the globals that we deem potentially confusing.
```
Comment (tdeekens) [inline packages/eslint-config-mc-app/helpers/rules-presets.js:9]: We enforce a deny list of the globals...

**Review by tdeekens — APPROVED** [review]
Good idea. Copy paste is often underrated :-). 

Just wondering if we want to periodically sync with upstream?

File: packages/eslint-config-mc-app/helpers/rules-presets.js (hunk 2 above)
Comment (emmenko) [reply packages/eslint-config-mc-app/helpers/rules-presets.js:9]: Yeah thanks, noticed that too.

**Pull request merged**